staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -73,6 +73,10 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 
 	chains, err := req.TLS.PeerCertificates[0].Verify(optsCopy)
 	if err != nil {
+		switch err.(type) {
+		case x509.UnknownAuthorityError:
```

Comment:
That's exactly what the current chain does. Letting any one item in the chain decide whether to short-circuit or not breaks composition